API/Controllers/BooksController.cs

```diff
@@ -0,0 +1,101 @@
+﻿using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using VL.Shared.Interfaces;
+using VL.Shared.Model;
+using static System.Reflection.Metadata.BlobBuilder;
+
+namespace API.Controllers
+{
+    [Authorize(AuthenticationSchemes = JwtBearerDefaults.AuthenticationScheme, Policy = "IsAllUsers")]
+    [Route("api/vl")]
+    [ApiController]
+    public class BooksController : ControllerBase
+    {
+        private readonly IBookService _bookService;
+
+        public BooksController(IBookService bookService)
+        {
+            _bookService = bookService;
+        }
+
+        /// <summary>
+        /// Get all books
+        /// </summary>
+        /// <returns>A list of all books</returns>
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesDefaultResponseType]
+        public async Task<IActionResult> GetBooks()
+        {
+            var books = await _bookService.GetBooksAsync();
+            if (books.Count == 0) return NotFound();
+            return Ok(books);
+        }
+
+        /// <summary>
+        /// Get a specific book by id
+        /// </summary>
+        /// <param name="id">id</param>
+        /// <returns>Book</returns>
+        [HttpGet("{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesDefaultResponseType]
+        public async Task<ActionResult<Book>> GetBook(int id)
+        {
+            var book = await _bookService.GetBookAsync(id);
+            if (book == null) return NotFound();
+            return Ok(book);
+        }
+
+        /// <summary>
+        /// Edit a book
+        /// </summary>
+        /// <param name="book">book</param>
+        /// <returns>Updated book</returns>
+        [HttpPut("{id}"), Authorize("IsLibrarian")]
+        [ProducesResponseType(StatusCodes.Status202Accepted)]
+        [ProducesDefaultResponseType]
+        public async Task<IActionResult> UpdateBook(UpdateBookDto book)
```

Comment:
If you're returning a type, you should specify it in your `Task<IActionResult<BookDto>`

Reply:
Won't allow me to without changing IActionResult to ActionResult. Please confirm ok.

Reply:
What I said was incorrect, when returning a type the concrete class should be used

`Task<ActionResult<BookDto>` would be right